src/deploy/functions/prompts.ts

```diff
@@ -120,3 +142,103 @@ export async function promptForFunctionDeletion(
   }
   return shouldDeleteFns;
 }
+
+/**
+ * Checks whether a deploy will increase the min instance idle time bill of
+ * any function. Cases include:
+ * * Setting minInstances on a new or existing function
+ * * Increasing the minInstances of an existing function
+ * * Increasing the CPU or memory of a function with min instances
+ * If there are any, prompts the user to confirm a minimum bill.
+ */
+export async function promptForMinInstances(
+  options: args.Options,
+  want: backend.FunctionSpec[],
+  have: backend.FunctionSpec[]
+): Promise<void> {
+  if (options.force) {
+    return;
+  }
+
+  const increasesCost = want.some((wantFn) => {
+    // If we don't know how much this will cost, be pessimal
+    if (!pricing.canCalculateMinInstanceCost(wantFn)) {
+      return true;
+    }
+    const wantCost = pricing.monthlyMinInstanceCost([wantFn]);
+    const haveFn = have.find(backend.sameFunctionName(wantFn));
+    let haveCost;
+    if (!haveFn) {
+      haveCost = 0;
+    } else if (!pricing.canCalculateMinInstanceCost(wantFn)) {
+      return true;
+    } else {
+      haveCost = pricing.monthlyMinInstanceCost([haveFn]);
+    }
+    return wantCost > haveCost;
+  });
+
+  if (!increasesCost) {
+    return;
+  }
+
+  if (options.nonInteractive) {
+    throw new FirebaseError(
+      "Pass the --force option to deploy functions that increase the minimum bill",
+      {
+        exit: 1,
+      }
+    );
+  }
+
+  // Conisderations for future versions:
+  // Group Tier 1 and Tier 2 regions
+  // Add Tier 1 or Tier 2 annotations to functionLines
+  const functionLines = want
+    .filter((fn) => fn.minInstances)
+    .sort(compareFunctions)
+    .map((fn) => {
+      return (
+        `\t${getFunctionLabel(fn)}: ${fn.minInstances} instances, ` +
+        backend.memoryOptionDisplayName(fn.availableMemoryMb || 256) +
+        " of memory each"
+      );
+    })
+    .join("\n");
+
+  let costLine;
+  if (want.some((fn) => !pricing.canCalculateMinInstanceCost(fn))) {
+    costLine =
+      "Cannot calculate the minimum monthly bill for this configuration. Consider running " +
+      clc.bold("npm install -g firebase-tools");
+  } else {
+    const cost = pricing.monthlyMinInstanceCost(want).toFixed(2);
+    costLine = `With these options, your minimum bill will be $${cost} in a 30-day month.`;
```

Comment:
This is because we want something like $0.60. The first $ is the literal $ printed and the second $ is for string interpolation. Don't worry, it works 🙃